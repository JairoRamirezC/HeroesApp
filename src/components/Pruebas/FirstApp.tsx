import { useState } from "react";
import {CounterApp} from "./CounterApp";


const FirstApp = () => {
  const [counter, setCounter] = useState<number>(0);
  const [titles, setTitles] = useState<string>('');

  const handleIncrementCounter = () => {
    setCounter(prev => prev + 1);
  }
  
  const handleDecrementCounter = () => {
    if(counter === 0) return;
    setCounter(prev => prev - 1);
  }

  return (
    <>
      <p>{counter}</p>

      <button onClick={handleIncrementCounter}>Sumar +1</button>
      <button onClick={handleDecrementCounter}>Restar -1</button>
      <button onClick={() => setTitles('Nuevo nombre')}>Cambiar nombre</button>
      <CounterApp title={`${titles ? titles : 'React Memo'}`} subTitle="Si como no" name="Jairo" />
    </>
  )
}

export default FirstApp