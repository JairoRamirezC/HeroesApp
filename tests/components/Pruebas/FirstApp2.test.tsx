import { render, screen } from "@testing-library/react";
import FirstApp from "../../../src/components/Pruebas/FirstApp";

describe('Testing FirstApp Improved', () => {
  const text = 'Cambiar nombre';

  test('FirstApp should match to the snapshot', () => {
    const {container} = render(<FirstApp />);
    expect(container).toMatchSnapshot();
  });

  test('FirstApp should render text "Cambiar nombre"', () => {
    render(<FirstApp />);
    const buttonElementText = screen.getByText(text);
    expect(buttonElementText).toBeTruthy();
  });

  test('counter should render 0 value into P element', () => {
    render(<FirstApp />);
    const pElementCounter = screen.getByText('0');
    expect(pElementCounter).toBeTruthy();
    expect(pElementCounter.tagName).toBe('P');
  });

  test('FirstApp should render text "Cambiar nombre" into button tag', () => {
    render(<FirstApp />);
    const buttonElement = screen.getAllByRole('button');
    expect(buttonElement[2]?.textContent).toContain(text);
  });

  test('button add should be have "Sumar +1" in the text', () => {
    const textButton = 'Sumar +1';
    render(<FirstApp />);
    const btnElement = screen.getAllByRole('button');
    expect(btnElement[0].textContent).toBe(textButton);
  });
})