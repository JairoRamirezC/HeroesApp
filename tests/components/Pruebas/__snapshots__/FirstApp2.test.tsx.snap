// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Testing FirstApp Improved FirstApp should match to the snapshot 1`] = `
<div>
  <p>
    0
  </p>
  <button
    data-testid="firstApp-btn-sumar"
  >
    Sumar +1
  </button>
  <button>
    Restar -1
  </button>
  <button>
    Cambiar nombre
  </button>
  <div>
    <h1>
      React Memo
    </h1>
    <p>
      Si como no
    </p>
    <p>
      Jairo
    </p>
  </div>
</div>
`;
